### PR TITLE
fix storage secret field omit empty

### PR DESCRIPTION
**Pull request by soulseen**
Signed-off-by: Xiaoyang Zhu <zhuxiaoyang1996@gmail.com>

### CODE REVIEW

File: controllers/storage/provision.go [hunk 1]
```diff
@@ -191,13 +169,7 @@ func (m *MinIOReconciler) generateAzureSecret(labels map[string]string) *corev1.
...
 
 func (m *MinIOReconciler) generateGcsSecret(labels map[string]string) *corev1.Secret {
-	data := map[string]string{
-		"bucket":        m.HarborCluster.Spec.Storage.Gcs.Bucket,
-		"encodedkey":    m.HarborCluster.Spec.Storage.Gcs.EncodedKey,
-		"rootdirectory": m.HarborCluster.Spec.Storage.Gcs.RootDirectory,
-		"chunksize":     m.HarborCluster.Spec.Storage.Gcs.ChunkSize,
-	}
-	dataJson, _ := json.Marshal(&data)
+	dataJson, _ := json.Marshal(m.HarborCluster.Spec.Storage.Gcs)
```
Comment (steven-zou) [inline controllers/storage/provision.go:172]: Is it ok to ignore the error here? Can we make sure the `Spec.Storage` is valid? By validation webhook?

@soulseen

Comment (steven-zou) [inline controllers/storage/provision.go:172]: Is it ok to ignore the error here? Can we make sure the Spec.Storage is valid? By validation webhook?

@soulseen

Comment (soulseen) [reply controllers/storage/provision.go:172]: I forgot to handle error here

File: controllers/database/generate.go [hunk 2]
```diff
@@ -45,6 +45,18 @@ func (postgres *PostgreSQLReconciler) generatePostgresCR() (*unstructured.Unstru
 				},
 				"foo_user": {},
 			},
+			DockerImage: "harbor.ymmoa.com/monitoring/postgresql:12",
```
Comment (Ghostbaby) [inline controllers/database/generate.go:48]: It's my mistake, pls help me remove this line. - -.
```suggestion
```

File: controllers/storage/provision.go [hunk 3]
```diff
@@ -158,17 +141,11 @@ func (m *MinIOReconciler) generateS3Secret(labels map[string]string) *corev1.Sec
...
-func (m *MinIOReconciler) generateAzureSecret(labels map[string]string) *corev1.Secret {
-	data := map[string]string{
-		"realm":       m.HarborCluster.Spec.Storage.Azure.Realm,
-		"accountname": m.HarborCluster.Spec.Storage.Azure.AccountName,
-		"accountkey":  m.HarborCluster.Spec.Storage.Azure.AccountKey,
-		"container":   m.HarborCluster.Spec.Storage.Azure.Container,
-	}
-	dataJson, _ := json.Marshal(&data)
+func (m *MinIOReconciler) generateAzureSecret(labels map[string]string) (*corev1.Secret, error) {
+	dataJson, err := json.Marshal(m.HarborCluster.Spec.Storage.Azure)
```
Comment (steven-zou) [inline controllers/storage/provision.go:148]: Maybe dive a little bit of detail, why not return `nil, err` here and `&corev1.Secret{}, nil` at the end? Want to reduce code lines?

**Pull request merged**